src/Data/Aeson/Types/Internal.hs

```diff
@@ -530,6 +547,71 @@ formatRelativePath path = format "" path
     escapeChar '\\' = "\\\\"
     escapeChar c    = [c]
 
+getFieldName :: JSONPath -> Maybe String
+getFieldName path =
+    format $ reverse path
+    where
+        format :: JSONPath -> Maybe String
+        format []                = Nothing
+        format (Index idx:parts) = format parts
+        format (Key key:_)   = Just $ formatKey key
+
+        formatKey :: Text -> String
+        formatKey key
+            | isIdentifierKey strKey = strKey
+            | otherwise              = escapeKey strKey
+            where strKey = unpack key
+
+        isIdentifierKey :: String -> Bool
+        isIdentifierKey []     = False
+        isIdentifierKey (x:xs) = isAlpha x && all isAlphaNum xs
+
+        escapeKey :: String -> String
+        escapeKey = concatMap escapeChar
+
+        escapeChar :: Char -> String
```

Comment:
can we make the IError JSONPath to include the field name without `/`. not necessary, but just check would that be more appropriate then doing a case match. because is it possible to get `\\\`